### PR TITLE
Add error boundary to monaco editors

### LINKED ISSUE #909: Use error boundaries in Busola
**Description**

Introduce "catch-all" style handlers in various places of Busola - especially for create/edit modals.Those handlers would show small message or handle the error gracefully instead of crashing the entire MF.

**Reasons**

Even though we validate our inputs, users can type anything into the YAML editor - and for example if we rely on some value being of particular type, then an error may arise when user types e.g. object instead of an array.

Remember to rethrow caught errors again, so that Sentry can catch them and display on dashboard - they are _not_ expected (or use Sentry functionality to capture the error, if such functionality exists).

### PR DESCRIPTION
**Pull request by wchrapka**
**Description**

Adds Error boundary component for monaco editors

**Related issue(s)**

Resolves #909

### CODE REVIEW

**Review by dariadomagala — CHANGES_REQUESTED** [review]
We don't have the boundaries for the 'add cluster' editor.

**Comment by dariadomagala** [comment]
Also, could we add a test (might be just a unit test) for that?

File: shared/components/ErrorBoundary/ErrorBoundary.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    // Update state so the next render will show the fallback UI.
+    return { hasError: true, error };
+  }
+
+  render() {
+    if (this.state.hasError) {
+      return (
+        <div role="alert">
+          <p className="fd-margin-top-bottom--sm">
+            {this.props.t('err-boundary.restored-initial-form')}
```
[the remark this answers is not in the document]
Comment (wchrapka) [reply shared/components/ErrorBoundary/ErrorBoundary.js:22]: option added

**Review by Wawrzyn321 — CHANGES_REQUESTED** [review]
Would you kindly fix this case? When I paste such "kubeconfig", the error boundary is triggered, but then the error show up anyway:
```apiVersion: v1
kind: Config
current-context: shoot--hasselhoff--kmain
contexts:
  - name: shoot--hasselhoff--kmain
    context:
      cluster: shoot--hasselhoff--kmain
      user: shoot--hasselhoff--kmain-token
clusters:
  - name: shoot--hasselhoff--kmain
    cluster:
      server: https://api.kmain.hasselhoff.shoot.canary.k8s-hana.ondemand.com
      certificate-authority-data: 'yadadada'
users: 2 
```

**Comment by wchrapka** [comment]
> Would you kindly fix this case? When I paste such "kubeconfig", the error boundary is triggered, but then the error show up anyway:
> 
> ```
> kind: Config
> current-context: shoot--hasselhoff--kmain
> contexts:
>   - name: shoot--hasselhoff--kmain
>     context:
>       cluster: shoot--hasselhoff--kmain
>       user: shoot--hasselhoff--kmain-token
> clusters:
>   - name: shoot--hasselhoff--kmain
>     cluster:
>       server: https://api.kmain.hasselhoff.shoot.canary.k8s-hana.ondemand.com
>       certificate-authority-data: 'yadadada'
> users: 2 
> ```

That's the expected behavior on dev server, I haven't observed this issue in other envrionments. Could you describe how to reproduce?

**Review by Wawrzyn321 — APPROVED** [review]

File: core/src/i18n/en.yaml [hunk 2]
```diff
@@ -862,6 +862,9 @@ domains:
   include:
     label: Include Domains
     placeholder: Domain that is allowed
+err-boundary:
+  go-back: Go Back
```
Comment (grego952) [inline core/src/i18n/en.yaml:866]:
```suggestion
  go-back: Go back
```

File: core/src/i18n/en.yaml [hunk 3]
```diff
@@ -862,6 +862,9 @@ domains:
...
+  restored-initial-form: Something went wrong, we restored the initial form for you.
```
Comment (grego952) [inline core/src/i18n/en.yaml:867]:
```suggestion
  restored-initial-form: Something went wrong We restored the initial form for you.
```

**Review by Wawrzyn321 — APPROVED** [review]

**Review by grego952 — APPROVED** [review]

**Pull request merged**